src/core/qgsmaplayer.cpp

```diff
@@ -205,8 +205,11 @@ QString QgsMapLayer::id() const
 void QgsMapLayer::setId( const QString &id )
 {
   QGIS_PROTECT_QOBJECT_THREAD_ACCESS
+  if ( id == mID )
+    return;
 
   mID = id;
+  emit idChanged( id );
```

Comment:
Can you update QgsMapLayer::readLayerXml to rely on setId( mne,text() ) instead of a mId = mne.text so we get the signal there too?

Reply:
Ok, done